### PR TITLE
upgrade github action

### PR DESCRIPTION
**Pull request by ycyaoxdu**
Signed-off-by: ycyaoxdu <yaoyuchen0626@163.com>

### CODE REVIEW

**Review by qiujian16 — COMMENTED** [review]
would you also help to check if we need to update kind?

**Comment by ycyaoxdu** [comment]
> would you also help to check if we need to update kind?

@qiujian16 latest setup-kind is 0.5.0, latest kind is 0.17.0, updated

File: .github/workflows/go-presubmit.yml [hunk 1]
```diff
@@ -117,7 +117,7 @@ jobs:
       - name: setup kind
         uses: engineerd/setup-kind@v0.5.0
```
Comment (qiujian16) [inline .github/workflows/go-presubmit.yml]: do we need to upgrade this version?

Comment (ycyaoxdu) [reply .github/workflows/go-presubmit.yml]: @qiujian16  v0.5.0 is latest https://github.com/engineerd/setup-kind/tags

Comment (qiujian16) [reply .github/workflows/go-presubmit.yml]: e2e fails..

**Pull request merged**